File: Projects/Project_07_3D_Elliptic/For_Students/matlab/warp_grid.m
%---------------------------------------------------------------------%
%This function warp the grid and plots it.
%Written by F.X. Giraldo on 4/2008
%           Department of Applied Mathematics
%           Naval Postgraduate School 
%           Monterey, CA 93943-5216
%---------------------------------------------------------------------%
function coord = warp_grid(coord,npoin)

%Warp Grid
for i=1:npoin
    x=coord(1,i);
    y=coord(2,i);
    z=coord(3,i);
    coord(1,i)=x + sin(pi * x) * sin(2 * pi * y) * sin(2 * pi * z) / 15;
    coord(2,i)=y + sin(2 * pi * x) * sin(pi * y) * sin(2 * pi * z) / 15;
    coord(3,i)=z + sin(2 * pi * x) * sin(2 * pi * y) * sin(pi * z) / 15;
end




      


      

File: Projects/Project_06_2D_Wave/For_Instructors/julia/julia_Local_Matrices/warp_grid.jl
#---------------------------------------------------------------------#
#This function warps a 2D grid
#Written by F.X. Giraldo on 5/2019
#           Department of Applied Mathematics
#           Naval Postgraduate School
#           Monterey; CA 93943-5216
#---------------------------------------------------------------------#
function warp_grid!(coord,Npoin,DFloat)

    for i=1:Npoin
        x, y = coord[1,i], coord[2,i]
        coord[1,i]=x + sin(π * x) * sin(2 * π * y) / 50
        coord[2,i]=y + sin(2 * π * x) * sin(π * y) / 50
    end

end
